### PR TITLE
feat(dropdown): new tests for the dropdown component

### PR DESCRIPTION
**Pull request by masoudmanson**
- [x] Adding new test cases for the Dropdown component
- [x] Fixing closeOnBlur bug: Dropdown closes when button is clicked with closeOnBlue set to true

### CODE REVIEW

File: packages/components/src/core/Dropdown/index.test.tsx [hunk 1]
```diff
@@ -40,4 +61,154 @@ describe("<Dropdown />", () => {
...
+    const args = {
+      ...Test.args,
+      closeOnBlur: true,
+    };
+
+    render(<Test {...args} />);
+
+    // Open the dropdown
+    const DropdownElement = screen.getByTestId("dropdown");
+    userEvent.click(DropdownElement);
```
Comment (liaprins) [inline packages/components/src/core/Dropdown/index.test.tsx:129]: Doe sit actually need to use `userEvent` here, or would `fireEvent` work now that it has `wait(500)` below?

File: packages/components/src/core/Dropdown/index.test.tsx [hunk 2]
```diff
@@ -40,4 +61,154 @@ describe("<Dropdown />", () => {
...
+
+    // Awaiting to find the STATUS_CANT_FIND_TEXT on the screen to
+    // make sure that the dropdown is opened
+    await screen.findByText(STATUS_CANT_FIND_TEXT);
+
+    // Click on the button again to close it
+    userEvent.click(DropdownElement);
```
Comment (liaprins) [inline packages/components/src/core/Dropdown/index.test.tsx:136]: Same question about `userEvent` here as above...?

**Review by liaprins — APPROVED** [review]
I added a couple of comments, but it looks good to me!

File: packages/components/src/core/Dropdown/index.test.tsx (hunk 1 above)
Comment (masoudmanson) [reply packages/components/src/core/Dropdown/index.test.tsx:129]: Actually, it doesn't work with fireEvent, and we kinda have to use userEvent in this case!

Comment (liaprins) [reply packages/components/src/core/Dropdown/index.test.tsx:129]: Got it!

**Pull request merged**